gems/pending/appliance_console/database_replication.rb

```diff
@@ -0,0 +1,128 @@
+require 'appliance_console/logging'
+require 'appliance_console/prompts'
+require 'pg'
+require 'English'
+
+module ApplianceConsole
+  class DatabaseReplication
+    include ApplianceConsole::Logging
+
+    REPMGR_CONFIG = '/etc/repmgr.conf'.freeze
+
+    attr_accessor :cluster_name, :node_number,
+                  :database_name, :database_user, :database_password,
+                  :primary_host, :standby_host
+
+    def ask_for_unique_cluster_node_number
+      self.node_number = ask_for_integer("number uniquely identifying this node in the replication cluster")
+    end
+
+    def ask_for_database_credentials
+      ask_for_cluster_database_credentials
+      self.primary_host = ask_for_ip_or_hostname("primary database hostname or IP address", primary_host)
+    end
+
+    def confirm(including_standby_host = nil)
+      clear_screen
+      say(<<-EOL)
+Replication Server Configuration
+
+        Cluster Node Number:        #{node_number}
+        Cluster Database Name:      #{database_name}
+        Cluster Database User:      #{database_user}
+        Cluster Database Password:  "********"
+        Cluster Primary Host:       #{primary_host}
+        EOL
+      say("        Standby Host:               #{standby_host}") unless including_standby_host.nil?
+
+      agree("Apply this Replication Server Configuration? (Y/N): ")
+    end
+
+    def ask_continue_if_already_configured?
```

Comment:
This method name isn't great. Maybe it would make more sense as two methods?
- `repmgr_configured?` which checks for the config file
and
- `confirm_reconfiguration` which asks the user if they want to reconfigure.

This way things are a lot more clear from the calling code.